crates/testlang/outputs/cargo/slang_testlang/src/generated/napi_interface/cst.rs

```diff
@@ -70,10 +70,45 @@ impl RuleNode {
             .into()
     }
 
+    #[napi(catch_unwind, js_name = "toJSON")]
```

Comment:
I wonder if we should use `napi(string_enum)` on `NodeType` as well, since it just appears as `0` or `1` in the debugger.

Reply:
Having just the number as the discriminant ensures the footprint is lower, however I hope that V8 JIT would intern these or use some small string opt if we used strings here, instead.

Reply:
I will add a note to double check this later :+1

Reply:
I think given that we already use it on `TokenKind` and `RuleKind`, not sure if this is going to be different here, and I think it is useful for DX.
However, not a blocker. I can follow up on it later if not.

Reply:
Added in https://github.com/NomicFoundation/slang/pull/908/commits/eb63cda656926efc38cb0382884f3a24a992750e and verified that the type is correct:

![image](https://github.com/NomicFoundation/slang/assets/3093213/74d093c2-d2b8-4e8a-8129-5acad170764d)